### PR TITLE
templatectl: Add/Remove items from templates.json via cli

### PR DESCRIPTION
**Pull request by PeterGrace**
This PR adds a package file for templatectl, which is a rust-based binary that interacts with the `/usr/share/remarkable/templates/templates.json` file.  It can add entries and remove entries from the file, sparing the end user the need to edit the json file manually.

Additionally, since it is a cli-based app and will be installable via toltec, we can then use it to package template files in toltec for installing to xochitl from the package repository.

### CODE REVIEW

File: package/templatectl/package [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+# Copyright (c) 2020 The Toltec Contributors
+# SPDX-License-Identifier: MIT
+
+pkgnames=(templatectl)
+pkgdesc="Tool to add/remove templates for xochitl"
+url=https://github.com/PeterGrace/templatectl
+pkgver=0.1.0
+timestamp=2020-12-24T12:32Z
```
Comment (Eeems) [inline package/templatectl/package]: https://github.com/toltec-dev/toltec/blob/stable/docs/package.md#timestamp
This should match the release timestamp from your repo
```suggestion
timestamp=2021-01-14T01:54Z
```

File: package/templatectl/package [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+# Copyright (c) 2020 The Toltec Contributors
+# SPDX-License-Identifier: MIT
+
+pkgnames=(templatectl)
+pkgdesc="Tool to add/remove templates for xochitl"
+url=https://github.com/PeterGrace/templatectl
+pkgver=0.1.0
```
Comment (Eeems) [inline package/templatectl/package]: https://github.com/toltec-dev/toltec/blob/stable/docs/package.md#pkgver
> Always include a dash-separated package revision number at the end of the version, resetting it to 1 when bumping the software version, and increasing it when making changes to the recipe itself.

```suggestion
pkgver=0.1.0-1
```

File: package/templatectl/package [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+build() {
+    # Fall back to system-wide config
+    rm .cargo/config
+
+    cargo build --release
+}
+
+package() {
+    install -D -m 755 -t "$pkgdir"/opt/bin \
+      "$srcdir"/target/armv7-unknown-linux-gnueabihf/release/templatectl
```
Comment (Eeems) [inline package/templatectl/package]: Our linter gets grumpy
```suggestion
        "$srcdir"/target/armv7-unknown-linux-gnueabihf/release/templatectl
```

**Review by Eeems — CHANGES_REQUESTED** [review]
This should be rebased and retargetted against the testing branch.

**Comment by PeterGrace** [comment]
closing this one and opening a fresh one

**Pull request closed without merging**